src/main/java/tr/edu/ogu/ceng/controller/StudentController.java

```diff
@@ -77,5 +78,15 @@ public ResponseEntity<StudentResponseDto> registerAsStudent(@RequestBody Student
 	public StudentDto getStudentByUserId(@PathVariable Long userId) {
 		return studentService.getStudentByUserId(userId);
 	}
+	
+	@GetMapping("/supervisor/{id}")
+	public Page<Student> getAllStudentsByFacultySupervisorId(@PathVariable(name = "id") Long faculty_supervisor_id,
```

Comment:
Endpoint dönüş değerinin Student yerine StudentResponseDto olması gerekmez mi? Aynı durum repository ve service katmanı için de geçerli olmalı.

Reply:
değiştim